server/reportGenerator/src/reportGen.ts

```diff
@@ -0,0 +1,152 @@
+		"UI elements behind a modal bottom sheet lack a visible scrim/filter whilst the bottom sheet is onscreen.  A visible scrim indicates to the user that these elements cannot be interacted with whilst the menu is displayed.",
+	];
+
+	let res = "<h2>Item " + index.toString() + "</h2>";
+	res += "<p>" + titles[algResult] + "<br>" + desc[algResult] + "</p>";
+	return res;
+}
+
+/**
+ * Generates a html report for a given image issue
+ *
+ * @param index - index of original image in batch
+ * @param filePath - filepath to original image
+ * @param algResultPath - filepath to heatmap outputted from OwlEyes algorithm
+ * @param algResult - issue type code
+ * @returns html string containing image, heatmap, and description
+ */
+function generateImgReport(
+	index: number,
+	filePath: string,
+	algResultPath: string,
+	algResult: number
+): string {
+	const titles: string[] = [
+		"General issue heatmap",
+		"Null value",
+		"Missing image",
+		"Component occlusion",
+		// no models for blurred screen or text overlap, so no description is needed
+	];
+	const desc: string[] = [
+		"Heatmap highlights all potential issues.",
+		"'NULL' text is being displayed, instead of the correct information.",
+		"A placeholder 'missing/broken image' symbol is displayed, instead of an intended image.",
+		"Text is overlapped or obscured by other components.",
+	];
+
+	let res = "<h2>Item " + index.toString() + "</h2>";
+	res += "<image src='" + filePath + "'>";
+	res += "<image src='" + algResultPath + "'>";
+	res += "<p>" + titles[algResult] + ".<br>" + desc[algResult] + "</p>";
+	return res;
+}
+
+/**
+ * Adds html report for a batch to s3 bucket, then adds the batch to a queue for file zipping.
+ * The report is saved in the bucket as <batch_id>/report.html
+ * DynamoDB data about a batch is assumed to be in the format defined here: https://github.com/Visual-Testing-for-Android-Apps/backend/issues/15#issuecomment-898450609
+ * with the exception of "batch_id" being "id" instead due to issues.
+ *
+ * @param event object containing information about the job. Its body attribute is a stringified json object containing a batch's id as jobKey
+ * @param context object containing information about the invocation, function, and execution environment
+ * @returns html string describing issues identified for each image and/or video
+ */
+export const generateReport = async (event: SQSEvent, context: AWSLambda.Context): Promise<any> => {
+	// fetch job results from database
+	const key: string = JSON.parse(event.Records[0].body).jobKey; // queue batchsize = 1, so Records[0].length = 1
+	const request: GetItemInput = {
+		TableName: process.env.JOB_TABLE as string,
+		Key: {
+			id: { S: key },
+		},
+	};
+	const dbRes = await getItem(request);
+
+	// construct HTML report contents
+	let res = "<!DOCTYPE html><html lang='en'><head></head><body><div>";
+	if (dbRes.Item != null) {
+		const files = dbRes.Item.files;
```

Comment:
`item = Converter.unmarshell(dbRes.Item)`. It convert dynamodb json back to javascript json. Then you can just access each property directly (`element.M.fileRef`  becomes `element.fileRef`)